src/StatusQ/Controls/StatusInput.qml

```diff
@@ -30,22 +30,43 @@ Item {
     property int charLimit: 0
     property string errorMessage: ""
     property list<StatusValidator> validators
+    property bool dirty: false
+    property int validationMode: StatusInput.ValidationMode.OnlyWhenDirty
+    enum ValidationMode {
+        OnlyWhenDirty, // validates input only after it has become dirty
+        Always // validates input even before it has become dirty
+    }
 
     property var errors: ({})
 
     function validate() {
+        if (!dirty && validationMode === StatusInput.ValidationMode.OnlyWhenDirty) {
+            return
+        }
+        statusBaseInput.valid = true
         if (validators.length) {
             for (let idx in validators) {
-                let result = validators[idx].validate(statusBaseInput.text)
+                let validator = validators[idx]
+                let result = validator.validate(statusBaseInput.text)
 
                 if (typeof result === "boolean" && result) {
-                    statusBaseInput.valid = true
+                    statusBaseInput.valid = statusBaseInput.valid && true
+                    delete errors[validator.name]
                 } else {
                     if (!errors) {
                         errors = {}
                     }
-                    errors[validators[idx].name] = result
-                    statusBaseInput.valid = false
+                    result.error = validator.error
+                    errors[validator.name] = result
+                    statusBaseInput.valid = statusBaseInput.valid && false
```

Comment:
Oh, nice catch!!

Reply:
Note: this semantically the same as `statusBaseInput.valid = false`, but figured this was more descriptive.